node/src/blockchain/blockchain_interface.rs

```diff
@@ -785,37 +791,34 @@ mod tests {
             event_loop_handle,
             TEST_DEFAULT_CHAIN,
         );
-
         let end_block_nbr = 1024u64;
-
         let result = subject
             .retrieve_transactions(
                 BlockNumber::Number(42u64.into()),
                 BlockNumber::Number(end_block_nbr.into()),
-                &Wallet::from_str("0x3f69f9efd4f2592fd70be8c32ecd9dce71c472fc").unwrap(),
+                &Wallet::from_str(&to).unwrap(),
             )
             .unwrap();
 
         let requests = test_server.requests_so_far();
-        let bodies: Vec<Value> = requests
+
+        let bodies: Vec<String> = requests
             .into_iter()
             .map(|request| serde_json::from_slice(&request.body()).unwrap())
-            .collect();
-        let bodies: Vec<String> = bodies
             .into_iter()
-            .map(|b| serde_json::to_string(&b).unwrap())
+            .map(|b: Value| serde_json::to_string(&b).unwrap())
             .collect();
         assert_eq!(
-            bodies,
-            vec![format!("[{{\"id\":0,\"jsonrpc\":\"2.0\",\"method\":\"eth_blockNumber\",\"params\":[]}},{{\"id\":1,\"jsonrpc\":\"2.0\",\"method\":\"eth_getLogs\",\"params\":[{{\"address\":\"0x384dec25e03f94931767ce4c3556168468ba24c3\",\"fromBlock\":\"0x2a\",\"toBlock\":\"0x400\",\"topics\":[\"0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef\",null,\"0x000000000000000000000000{}\"]}}]}}]", &to[2..])],
+            bodies[0],
```

Comment:
I used your recommended changes but could not do so without introducing an `expected_body_suffix` for the closing JSON syntax. It would be nice if `format!()` could work with raw Strings.